diff --git a/include/mbedtls/mbedtls_config.h b/include/mbedtls/mbedtls_config.h
index bd3f71d5b..52e78efc9 100644
--- a/include/mbedtls/mbedtls_config.h
+++ b/include/mbedtls/mbedtls_config.h
@@ -302,6 +302,10 @@
  */
 //#define MBEDTLS_PLATFORM_ZEROIZE_ALT
 
+#define MBEDTLS_KEY_EXCHANGE_ECDH_ANON_ENABLED
+#define MBEDTLS_ALLOW_PRIVATE_ACCESS
+#define MBEDTLS_ENTROPY_MAX_SOURCES                2 /**< Maximum number of sources supported */
+
 /**
  * \def MBEDTLS_DEPRECATED_WARNING
  *
