diff --git a/Makefile.cflags b/Makefile.cflags
index bef46ca..560cff7 100644
--- a/Makefile.cflags
+++ b/Makefile.cflags
@@ -35,7 +35,7 @@ endif
 # Worse yet they hide errors by accepting wildcard argument types.
 ifeq ($(shell $(CC) -Wstrict-prototypes -Werror=strict-prototypes -Wold-style-definition -Werror=old-style-definition -E - 2>/dev/null >/dev/null </dev/null ; echo $$?),0)
   # duplicated parameters don't hurt
-  CFLAGS += -Wstrict-prototypes -Werror=strict-prototypes -Wold-style-definition -Werror=old-style-definition
+#  CFLAGS += -Wstrict-prototypes -Werror=strict-prototypes -Wold-style-definition -Werror=old-style-definition
   CXXUWFLAGS += -Wstrict-prototypes -Wold-style-definition
 endif
 
diff --git a/Makefile.include b/Makefile.include
index d9daae0..a0f0bc9 100644
--- a/Makefile.include
+++ b/Makefile.include
@@ -93,7 +93,7 @@ endif
 WERROR ?= 1
 export WERROR
 ifeq ($(WERROR),1)
-  CFLAGS += -Werror
+#  CFLAGS += -Werror
 endif
 
 ifneq (10,$(if ${RIOT_VERSION},1,0)$(if ${__RIOTBUILD_FLAG},1,0))
diff --git a/boards/native/Makefile.include b/boards/native/Makefile.include
index 9795b4d..cedf53c 100644
--- a/boards/native/Makefile.include
+++ b/boards/native/Makefile.include
@@ -44,7 +44,7 @@ export CGANNOTATE ?= cg_annotate
 export GPROF ?= gprof
 
 # basic cflags:
-export CFLAGS += -Wall -Wextra -pedantic
+#export CFLAGS += -Wall -Wextra -pedantic
 ifeq ($(shell uname -m),x86_64)
 export CFLAGS += -m32
 endif
