--- a/Arduino-Makefile/Arduino.mk	2019-04-11 15:28:21.579786532 +0200
+++ b/Arduino-Makefile/Arduino.mk	2019-04-11 15:09:39.302742785 +0200
@@ -889,7 +889,7 @@
 LOCAL_OBJ_FILES = $(LOCAL_C_SRCS:.c=.c.o)   $(LOCAL_CPP_SRCS:.cpp=.cpp.o) \
 		$(LOCAL_CC_SRCS:.cc=.cc.o)   $(LOCAL_PDE_SRCS:.pde=.pde.o) \
 		$(LOCAL_INO_SRCS:.ino=.ino.o) $(LOCAL_AS_SRCS:.S=.S.o)
-LOCAL_OBJS      = $(patsubst %,$(OBJDIR)/%,$(LOCAL_OBJ_FILES))
+LOCAL_OBJS      = $(patsubst %,$(OBJDIR)/%,$(notdir $(LOCAL_OBJ_FILES)))
 
 ifeq ($(words $(LOCAL_SRCS)), 0)
     $(error At least one source file (*.ino, *.pde, *.cpp, *c, *cc, *.S) is needed)
